designs/native-ami-versioning.md

```diff
@@ -0,0 +1,121 @@
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiSelector:
+    minimumAgeDuration: 10d  // metav1.Duration
+```
+
+### How `minimumAgeDuration` interacts with the other methods of AMI selection
+
+#### *EKS Optimized AMI (no other AMI Selector field)*
+
+1. Karpenter will compare `minimumAgeDuration` to when the SSM Parameter was last modified instead of using the Image Creation Date that's returned in `aws ec2 describe-images`. This is because the SSM Parameter is updated independently of image creation.
+2. The newest EKS Optimized AMI that fits the `minimumAgeDuration` selector will be chosen.
+
+#### *Arbitrary tags*
+
+1. An AMI tagged and discovered via the amiSelector tags will not be chosen unless it fits the `minimumAgeDuration` filter.
+2. This has the downside of resulting in no AMIs found if users are too restrictive with tags and the `minimumAgeDuration` filter. The upside is that users will know any AMI created before x days ago will not be included.
+
+#### *AMI IDs*
+
+1. When using AMI IDs, other tags specified are not respected. To keep explicit AMI ID usage as an override, using `minimumAgeDuration` and AMI IDs will result in `minimumAgeDuration` not being respected.
+
+### DriftTTL in the ProvisionerSpec
+
+Karpenter could include a drift TTL in the Provisioner. Nodes would have to be drifted for the TTL's duration to be deprovisioned. This can implement bake time, but will also apply for all future methods of Drift, such as zonal and instance permission requirements. Since other Drift methods may need more immediate actions when it occurs, setting a Drift TTL may not be granular as needed.
+
+### 🔑 Suggestion
+
+Implement Bake Time with `minimumAgeDuration`. Users can take advantage of it with EKS Optimized AMIs generally to follow requirements on AMI bake time. This applies only to AMIs, rather than a Drift TTL which would apply broadly to every provisioning requirement. Users can rely on `minimumAgeDuration` to use it as a safeguard against unexpected AMI upgrades, creating more confidence in relying on Karpenter to make the right decisions.
+
+## Release Pinning
+
+*🔑 Expand amiFamily to allow a pinned version for EKS Optimized AMIs.*
+
+Add an option to specify AMI Release Version with AMI Family. A user can pin known AMI Versions, and declaratively upgrade to a different AMI when ready. This could prevent automatic updates in the event of a newer EKS Optimized AMI. Some users [have requested this](https://github.com/aws/karpenter/issues/1495) with 13 up-votes.
+
+### *Option 1*
+
+Use an AMI Family with an appended magic string. This allows users who only know of version strings to modify AWSNodeTemplates easily to their liking.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiFamily: AL2-v20220824 # No version suffix defaults to latest
```

Comment:
While this is a cool semantic that maps to container image semantics (e.g. amiFamily: AL2:v20220824), I think we're mixing up concepts here.

amiFamily is currently a runtime contract that tells Karpenter what to expect for the user data, ebs volumes, etc. Users can specify amiFamily as well as specific amiIDs and this semantic would enable them to specify both simultaneously. 

Does it make sense to support pinning through amiID instead?

Reply:
> Does it make sense to support pinning through amiID instead?
as amiIDs are different in different regions (and for different architectures, gpu / non-gpu setups, etc), if we allow to pin only using amiID, we force users to perform tedious and error-prone lookups (while we could allow to specify some selected optional, potentially even amiFamily-specific constraints)

allowing to specify a version could be a great addition to the `minimumAgeDuration` bake time
(not saying that specifying these optional constraints in the `amiFamily` field itself is the best option)